### PR TITLE
Improved DAO stats + force cache flush + partial OptimisticLockException fix

### PR DESCRIPTION
**Pull request by erocs**
This partially fixes the OptimisticLockException you were seeing. I did see it happen once more with this code though so I need to track down that edge case.

New console commands. Of interest "ctcon daocachestats" and "ctcon daocachestatsslow". The latter could be slow when there are large numbers of reinforcements cached so it has a separate command. Also "ctcon forcecacheflush" will flush 5 caches to the DB. You can add a parameter to change the count, like "ctcon forcecacheflush 100", but the higher the number the longer it could take.

### CODE REVIEW

**Comment by erocs** [comment]
http://www.erocs.org/~erocs/Citadel_i82.jar

**Comment by ttk2** [comment]
Could we get those commands in a more humanly reasonable to to type format with a ELI5 explanation for when using them is a good idea and when it may end badly?

**Comment by erocs** [comment]
Sorry, didn't see this comment. Yes now that the cache is stable I can go back and pretty this up however you want.
